Updated Customscriptextension, docker service reconfigure

ewok-number-3 opened the pull request:
The Docker service on the agents has been stopping after creation, an extra command has been added to retry and restart the service. If that fails after two attements it will shutdown the agent. This will cause a new one to be spun up.

ewok-number-3 committed:
Updated Customscriptextension, docker service reconfigure

The Docker service on the agents has been stopping after creation, an extra command has been added to retry and restart the service. If that fails after two attements it will shutdown the agent. This will cause a new one to be spun up.

elduddz commented on customscriptextension.ps1:
```diff
@@ -73,6 +73,9 @@ Expand-Archive -Path ./$zip -DestinationPath .
 .\config.cmd --unattended --url https://dev.azure.com/$account --auth PAT --token $PAT --pool "$PoolName" --agent "$agentName" --runAsService
 
+Write-Information "###### Docker Service Config ######"
+SC failure docker reset=0 actions=restart/60000/restart/60000/run/60000 command=""shutdown" "/T00""
```
```suggestion
SC failure docker reset=0 actions=restart/60000/restart/120000/run/240000 command=""shutdown" "/T00""
```
$varService = "docker"
Get-Service -Name $varService | Restart-Service

ewok-number-3 replied:
Add this after your changes so if the service is not running it starts it else restart the service?

elduddz replied:
would need to use $Service in the logic, It might be ok to just do a Get-Service $varService | Restart-Service

if the service isnt running it will just start it.

elduddz reviewed (approved):
might be interesting to stretch it out between each attempt?

elduddz commented:
Sorry had a thought, probably want to check the service is started

elduddz reviewed (changes requested):
Thinking checking if the service is stopped would be good?

ewok-number-3 commented:
> Thinking checking if the service is stopped would be good?

I will have a look now.

ewok-number-3 committed:
Update Script, to restart service after config changes to it

79 Write-Information "###### Docker Service Restart ######"
80 $varService = "docker"
81 Get-Service -Name $varService | Restart-Service

ewok-number-3 commented:
The build will not complete due to the file being excluded
